core/app/models/spree/calculator/weighted_rate.rb

```diff
@@ -0,0 +1,19 @@
+require_dependency 'spree/calculator'
+
+module Spree
+  class Calculator::WeightedRate < Calculator
```

Comment:
Or maybe `DistributedAmount` (singular instead of plural, since the amount specified to the calculator is a single amount?)